pkg/synthetictests/allowedalerts/watchdog.go

```diff
@@ -72,7 +75,47 @@ func IsWatchdogAlert(eventInterval monitorapi.EventInterval) bool {
 	return eventInterval.Locator == "alert/Watchdog ns/openshift-monitoring"
 }
 
+func isSNOUpgradeTest(ctx context.Context, restConfig *rest.Config) (bool, error) {
+	configClient, err := configclient.NewForConfig(restConfig)
+	if err != nil {
+		return false, fmt.Errorf("Failed to establish API server connection")
+	}
+	infra, err := configClient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Failed to find Infra config")
+	}
+	// Exit early if its not SNO
+	if infra.Status.ControlPlaneTopology != configv1.SingleReplicaTopologyMode {
+		return false, nil
+	}
+
+	clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Failed to find cluster version")
+	}
+	// Make sure this test runs on upgrade
```

Comment:
aren't you really trying to ensure the test does *not* run on upgrade?  i think this comment is more accurately "check if this cluster is being upgraded"